extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/reactive/ReactiveMongoCollection.java

```diff
@@ -0,0 +1,1410 @@
+            ChangeStreamOptions options);
+
+    /**
+     * Creates a change stream for this collection.
+     *
+     * @param clientSession the client session with which to associate this operation
+     * @param options the stream options
+     * @return the stream of changes
+     */
+    Multi<ChangeStreamDocument<Document>> watch(ClientSession clientSession,
+            ChangeStreamOptions options);
+
+    /**
+     * Creates a change stream for this collection.
+     *
+     * @param clientSession the client session with which to associate this operation
+     * @param clazz the class to decode each document into
+     * @param <D> the target document type of the iterable.
+     * @param options the stream options
+     * @return the stream of changes
+     */
+    <D> Multi<ChangeStreamDocument<D>> watch(ClientSession clientSession, Class<D> clazz,
+            ChangeStreamOptions options);
+
+    /**
+     * Creates a change stream for this collection.
+     *
+     * @param clientSession the client session with which to associate this operation
+     * @param pipeline the aggregation pipeline to apply to the change stream
+     * @param options the stream options
+     * @return the stream of changes
+     */
+    Multi<ChangeStreamDocument<Document>> watch(ClientSession clientSession, List<? extends Bson> pipeline,
+            ChangeStreamOptions options);
+
+    /**
+     * Creates a change stream for this collection.
+     *
+     * @param clientSession the client session with which to associate this operation
+     * @param pipeline the aggregation pipeline to apply to the change stream
+     * @param clazz the class to decode each document into
+     * @param <D> the target document type of the iterable.
+     * @param options the stream options
+     * @return the stream of changes
+     */
+    <D> Multi<ChangeStreamDocument<D>> watch(ClientSession clientSession, List<? extends Bson> pipeline,
+            Class<D> clazz, ChangeStreamOptions options);
+
+    Multi<Document> mapReduce(String mapFunction, String reduceFunction);
+
+    <D> Multi<D> mapReduce(String mapFunction, String reduceFunction, Class<D> clazz);
+
+    Multi<Document> mapReduce(ClientSession clientSession, String mapFunction, String reduceFunction);
+
+    <D> Multi<D> mapReduce(ClientSession clientSession, String mapFunction, String reduceFunction,
+            Class<D> clazz);
+
+    Multi<Document> mapReduce(String mapFunction, String reduceFunction, MapReduceOptions options);
+
+    <D> Multi<D> mapReduce(String mapFunction, String reduceFunction, Class<D> clazz,
+            MapReduceOptions options);
+
+    Multi<Document> mapReduce(ClientSession clientSession, String mapFunction, String reduceFunction,
+            MapReduceOptions options);
+
+    <D> Multi<D> mapReduce(ClientSession clientSession, String mapFunction, String reduceFunction,
+            Class<D> clazz,
+            MapReduceOptions options);
```

Comment:
Missing JavaDoc, I know the original ReactiveMongoCollection didn't have JavaDoc for these methods but as they are part of the API and documented inside the MongoClient they should have some.

Reply:
Agreed!

Reply:
I added the missing JavaDoc.